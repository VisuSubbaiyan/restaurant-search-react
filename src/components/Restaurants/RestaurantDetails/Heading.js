import React from 'react';

const Heading = ({
  name,
  url,
  rating,
  location = {},
  phone,
  display_phone
}) => (
    <div className="heading">
      <h2>
        <a
          href={url}
          title={name}
          target="_blank"
          rel="noopener noreferrer"
        >
          {name}
        </a>
      </h2>
      <div><span className="rating">Rating: </span><span>{rating}</span></div>
      {location.formatted_address && <address className="full-address">{location.formatted_address}</address>}
      {display_phone && <a href={`tel:${phone}`} className="telephone">{display_phone}</a>}
    </div>
  );

export default Heading;
