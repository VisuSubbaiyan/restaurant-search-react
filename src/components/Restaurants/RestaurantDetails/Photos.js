import React from 'react';

const Photos = ({ name, photos: [photo] }) => (
  <div className="photo">
    <img src={photo} alt={name} />
  </div>
);

Photos.defaultProps = {
  photos: []
};

export default Photos;
