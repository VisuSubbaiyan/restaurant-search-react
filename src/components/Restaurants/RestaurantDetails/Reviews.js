import React from 'react';

const Reviews = ({ reviews = [] }) => (
  <div className="reviews">
    <h3>Reviews</h3>
    {reviews.length > 0 ?
      (<ul>
        {reviews.map((review, index) => (
          <li key={`${index}-${review.text}`}>
            <div>
              <span className="rating">Rating: </span>
              <span>{review.rating}</span>
            </div>
            <p>{review.text}</p>
          </li>
        ))}
      </ul>) : 'No reviews'}
  </div>
);

export default Reviews;
