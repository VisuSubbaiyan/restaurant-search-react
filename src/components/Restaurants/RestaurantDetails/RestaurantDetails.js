import React from 'react';
import PropTypes from 'prop-types';
import { pick } from 'lodash';

import Heading from './Heading';
import Photos from './Photos';
import Reviews from './Reviews';

const RestaurantDetails = props => {
  const headingProps = pick(props,
    ['name', 'url', 'location', 'phone', 'display_phone', 'rating']
  );

  return (
    <div className="restaurant-details">
      <Heading {...headingProps} />
      <Photos name={props.name} photos={props.photos} />
      <Reviews reviews={props.reviews} />
    </div>
  );
};

RestaurantDetails.propTypes = {
  name: PropTypes.string,
  location: PropTypes.object,
  phone: PropTypes.string,
  photos: PropTypes.arrayOf(PropTypes.string),
  reviews: PropTypes.arrayOf(PropTypes.object)
};

export default React.memo(RestaurantDetails);
