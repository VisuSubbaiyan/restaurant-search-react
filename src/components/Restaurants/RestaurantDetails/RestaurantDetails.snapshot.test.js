import React from 'react';

import RestaurantDetails from './RestaurantDetails';
import { createMountedComponent, sampleData } from '../../../utils/testUtils';

describe('<RestaurantDetails /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <RestaurantDetails {...sampleData.restaurants[0]} />
  });

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
  })
});
