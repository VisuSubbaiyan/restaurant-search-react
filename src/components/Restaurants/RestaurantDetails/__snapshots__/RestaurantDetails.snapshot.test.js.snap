// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantDetails /> should render correctly according to props 1`] = `
"<div class=\\"restaurant-details\\">
  <div class=\\"heading\\">
    <h2><a href=\\"https://www.yelp.com/biz/billy-the-butcher-hamburg-2?adjust_creative=nMAD8ywU1QBl_RMSM215Bg&amp;utm_campaign=yelp_api_v3&amp;utm_medium=api_v3_graphql&amp;utm_source=nMAD8ywU1QBl_RMSM215Bg\\" title=\\"Billy The Butcher\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Billy The Butcher</a></h2>
    <div><span class=\\"rating\\">Rating: </span><span>4</span>
    </div><address class=\\"full-address\\">Jungfernstieg 16 - 20
4. OG
20354 Hamburg
Germany</address><a href=\\"tel:+494057283299\\" class=\\"telephone\\">+49 40 57283299</a>
  </div>
  <div class=\\"photo\\">
    <img src=\\"https://s3-media2.fl.yelpcdn.com/bphoto/RDKmsxZZczP4EiTCB9AYOw/o.jpg\\"
    alt=\\"Billy The Butcher\\">
  </div>
  <div class=\\"reviews\\">
    <h3>Reviews</h3>
    <ul>
      <li>
        <div><span class=\\"rating\\">Rating: </span><span>4</span>
        </div>
        <p>I've had decent lunches at Billy The Butcher a couple of times now. It's
          what I would describe as a trendy burger joint on the 4th floor of the
          Alsterhaus...</p>
      </li>
      <li>
        <div><span class=\\"rating\\">Rating: </span><span>4</span>
        </div>
        <p>We arrived Hamburg at noon and decide to get some burgers before going
          to the office. This place is pretty close to the city hall and rated well.
          I ordered...</p>
      </li>
      <li>
        <div><span class=\\"rating\\">Rating: </span><span>4</span>
        </div>
        <p>Nice place to have good burger for lunch. Most of personnel speak English
          which is good. Prices are decent, also I would recommend to share fries
          with a...</p>
      </li>
    </ul>
  </div>
</div>"
`;
